src/admin/index.js

```diff
@@ -772,6 +775,280 @@ module.exports = ({
     })
   }
 
+  /**
+   * @param {array} ACL
+   * @return {Promise<void>}
+   */
+  const createAcls = async ({ acl }) => {
+    if (!acl || !Array.isArray(acl)) {
+      throw new KafkaJSNonRetriableError(`Invalid ACL array ${acl}`)
+    }
+    if (acl.length === 0) {
+      throw new KafkaJSNonRetriableError(`Empty ACL array`)
+    }
+
+    // Validate principal
+    if (acl.filter(({ principal }) => typeof principal !== 'string').length > 0) {
+      throw new KafkaJSNonRetriableError(
+        'Invalid ACL array, the principals have to be a valid string'
+      )
+    }
+    // Validate host
+    if (acl.filter(({ host }) => typeof host !== 'string').length > 0) {
+      throw new KafkaJSNonRetriableError('Invalid ACL array, the hosts have to be a valid string')
+    }
+    // Validate resourceName
+    if (acl.filter(({ resourceName }) => typeof resourceName !== 'string').length > 0) {
+      throw new KafkaJSNonRetriableError(
+        'Invalid ACL array, the resourceNames have to be a valid string'
+      )
+    }
+    // Validate operation
+    if (acl.filter(({ operation }) => typeof operation !== 'number').length > 0) {
```

Comment:
Rather than checking type, can we check if it is a member of the operations? Same for resourcePatternType, permissionType and resourceType. It's quite hard for the user to understand what is a valid operation if all they are told is that it has to be a number, rather than telling them: "Operation should be one of [1,2,3]"